### PR TITLE
mgmt: mcumgr: fs_mgmt: Allow querying supported hash/checksums

### PR DESCRIPTION
**Pull request by nordicjm**
Adds a new mcumgr command for returning information on all supported hash/checksum types that the firmware supports.

- [X] Code added
- [X] Release notes updated
- [X] SMP documentation updated
- [X] Manually tested
- [X] Test created and added to zephyr (WIP)
- [X] Automated test passes

### CODE REVIEW

**Comment by nordicjm** [comment]
CC @oyvindronningstad

**Review by carlescufi — DISMISSED** [review]
Looks great to me!

**Comment by nordicjm** [comment]
Rebased

**Review by oyvindronningstad — DISMISSED** [review]
The zcbor-related additions LGTM

File: subsys/mgmt/mcumgr/lib/cmd/fs_mgmt/src/fs_mgmt.c [hunk 1]
```diff
@@ -583,6 +640,10 @@ static const struct mgmt_handler fs_mgmt_handlers[] = {
 		.mh_read = fs_mgmt_file_hash_checksum,
 		.mh_write = NULL,
 	},
+	[FS_MGMT_ID_SUPPORTED_HASH_CHECKSUM] = {
+		.mh_read = fs_mgmt_supported_hash_checksum,
+		.mh_write = NULL,
+	},
```
[the remark this answers is not in the document]
Comment (nordicjm) [reply subsys/mgmt/mcumgr/lib/cmd/fs_mgmt/src/fs_mgmt.c:647]: Fixed

File: subsys/mgmt/mcumgr/lib/cmd/fs_mgmt/include/fs_mgmt/hash_checksum_mgmt.h [hunk 2]
```diff
@@ -73,6 +83,14 @@ void hash_checksum_mgmt_unregister_group(struct hash_checksum_mgmt_group *group)
  */
 const struct hash_checksum_mgmt_group *hash_checksum_mgmt_find_handler(const char *name);
 
+/**
+ * @brief Runs a callback with all supported hash/checksum types.
+ *
+ * @param cb		The callback function to call with each hash/checksum type.
+ * @param user_data	Data to pass back with the callback function.
+ */
+void hash_checksum_mgmt_find_handlers(hash_checksum_mgmt_list_cb cb, void *user_data);
```
[the remark this answers is not in the document]
Comment (nordicjm) [reply subsys/mgmt/mcumgr/lib/cmd/fs_mgmt/include/fs_mgmt/hash_checksum_mgmt.h:92]: Will need to rename them all en masse in a new PR

[the remark this answers is not in the document]
Comment (de-nordic) [reply subsys/mgmt/mcumgr/lib/cmd/fs_mgmt/include/fs_mgmt/hash_checksum_mgmt.h:92]: > Will need to rename them all en masse in a new PR

Yes, that is what I mean. You can log an issue for that.

[the remark this answers is not in the document]
Comment (nordicjm) [reply subsys/mgmt/mcumgr/lib/cmd/fs_mgmt/include/fs_mgmt/hash_checksum_mgmt.h:92]: Issue opened at https://github.com/zephyrproject-rtos/zephyr/issues/52099

File: doc/services/device_mgmt/smp_groups/smp_group_8.rst [hunk 3]
```diff
@@ -445,3 +447,80 @@ where:
     +-----------------------+---------------------------------------------------+
 
 In case when "rc" is not 0, success, the other fields will not appear.
+
+Supported file hash/checksum types
+**********************************
+
+Command allows listing which hash and checksum types are available on a device.
+Requires Kconfig :kconfig:option:`CONFIG_FS_MGMT_CHECKSUM_HASH_SUPPORTED_CMD`
```
Comment (de-nordic) [inline doc/services/device_mgmt/smp_groups/smp_group_8.rst:455]: Use the `CONFIG_MCUMGR_GRP_FS_CHECKSUM_HASH_SUPPORTED_CMD`, to keep the new Kconfig option format for MCUmgr, as You have already did here https://github.com/zephyrproject-rtos/zephyr/pull/50578/files#diff-834286adf361cd54c1b8c09197543580009a48bd6988bc78240a2487ba731f0eR91

File: subsys/mgmt/mcumgr/lib/cmd/fs_mgmt/Kconfig [hunk 4]
```diff
@@ -115,6 +115,21 @@ config FS_MGMT_HASH_SHA256
 	help
 	  Enable SHA256 hash support for mcumgr.
 
+config FS_MGMT_CHECKSUM_HASH_SUPPORTED_CMD
```
Comment (de-nordic) [inline subsys/mgmt/mcumgr/lib/cmd/fs_mgmt/Kconfig:118]: Use the `CONFIG_MCUMGR_GRP_FS_CHECKSUM_HASH_SUPPORTED_CMD`, to keep the new Kconfig option format for MCUmgr, as You have already did here https://github.com/zephyrproject-rtos/zephyr/pull/50578/files#diff-834286adf361cd54c1b8c09197543580009a48bd6988bc78240a2487ba731f0eR91

**Review by de-nordic — CHANGES_REQUESTED** [review]

File: subsys/mgmt/mcumgr/lib/cmd/fs_mgmt/Kconfig (hunk 4 above)
Comment (nordicjm) [reply subsys/mgmt/mcumgr/lib/cmd/fs_mgmt/Kconfig]: Fixed

File: doc/services/device_mgmt/smp_groups/smp_group_8.rst (hunk 3 above)
Comment (nordicjm) [reply doc/services/device_mgmt/smp_groups/smp_group_8.rst]: Fixed

**Review by de-nordic — APPROVED** [review]
Looks OK. Thanks.

**Review by carlescufi — APPROVED** [review]

**Pull request merged**